cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -269,7 +269,9 @@ impl Environment for ManagedEnvironment {
             .add_generation(&mut local_checkout, metadata)
             .map_err(ManagedEnvironmentError::CommitGeneration)?;
         self.lock_pointer()?;
-        local_checkout.link(flox, &self.out_link, &result.store_path)?;
+        if let Some(ref store_path) = &result.store_path {
```

Comment:
is there a reason why `install`, etc. would not return a store path, i.e. `None`?

Reply:
From the commit message
```
As far as I can tell, the only times that `store_path` would be `None`
is when we've attempted a transaction on a temporary environment and
found that we don't need the resulting build, such as when a package is
already installed.
```

Reply:
I guess that's also why `transact_with_manifest_contents` and `replace_with` explicitly don't update out-links?

Changing the `*Attempt` structs to carry the `new_manifest` and `store_path` together as a single `Option<>` would probably make it clearer but I'm going to cut my losses for now.